mapproxy/cache/tile.py

```diff
@@ -362,17 +376,24 @@ def _create_single_tile(self, tile):
                          self.tile_mgr.request_format, dimensions=self.dimensions)
         with self.tile_mgr.lock(tile):
             if not self.is_cached(tile):
-                source = self._query_sources(query)
-                if not source: return []
-                if self.tile_mgr.image_opts != source.image_opts:
-                    # call as_buffer to force conversion into cache format
-                    source.as_buffer(self.tile_mgr.image_opts)
-                source.image_opts = self.tile_mgr.image_opts
-                tile.source = source
-                tile.cacheable = source.cacheable
-                tile = self.tile_mgr.apply_tile_filter(tile)
-                if source.cacheable:
-                    self.cache.store_tile(tile)
+                try:
+                    source = self._query_sources(query)
```

Comment:
perhaps it makes sense to try-catch only this one statement? everything else does not throw such exception

Reply:
Yes, that's a good idea, it also makes sense with further work on https://github.com/camptocamp/mapproxy/pull/1

Reply:
done -> https://github.com/mapproxy/mapproxy/pull/518/commits/20db92c1084b9d115f5c863c65faf0fdefb19149